### PR TITLE
Updated the revised main.go code

### PR DESCRIPTION
**Pull request by chaithra-shivaprasad**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -145,19 +145,27 @@ func tokenProxyHandler(tokenEndpoint, repoPrefix string) http.HandlerFunc {
 		FlushInterval: -1,
 		Director: func(r *http.Request) {
 			orig := r.URL.String()
+                        log.Printf("In this function")
```
Comment (alextylor-e2x) [inline main.go]: debugging, we can remove

File: main.go [hunk 2]
```diff
@@ -145,19 +145,27 @@ func tokenProxyHandler(tokenEndpoint, repoPrefix string) http.HandlerFunc {
...
 
 			q := r.URL.Query()
 			scope := q.Get("scope")
-			if scope == "" {
-				return
+                        log.Printf("This is repoPrefix: %s", repoPrefix)
+                        repos := strings.Split(repoPrefix, "/")
+                        if ( scope == ""){
+                           scope = "repository:" + repos[1] + "/" + repos[1] + ":pull"
```
Comment (alextylor-e2x) [inline main.go]: move log into if block, print what we are setting scope too:
```
log.Printf("Setting missing scope to: %s", scope)
```
(after we set scope, so we can see the change)

also, update setting scope var to more canonical GO way, i.e. fmt.Sprintf(...)

File: main.go [hunk 3]
```diff
@@ -145,19 +145,27 @@ func tokenProxyHandler(tokenEndpoint, repoPrefix string) http.HandlerFunc {
...
 			}
-			newScope := strings.Replace(scope, "repository:", fmt.Sprintf("repository:%s/", repoPrefix), 1)
-			q.Set("scope", newScope)
+                        newScope := strings.Replace(scope, "repository:", fmt.Sprintf("repository:%s/", repoPrefix), 1)
+			q.Set("scope", newScope) 
+                        log.Printf("this is the scope: %s", scope)
```
Comment (alextylor-e2x) [inline main.go]: can remove this log line as debugging, and behaviur change comes form new log in if block

File: main.go [hunk 4]
```diff
@@ -145,19 +145,27 @@ func tokenProxyHandler(tokenEndpoint, repoPrefix string) http.HandlerFunc {
...
 			u, _ := url.Parse(tokenEndpoint)
 			u.RawQuery = q.Encode()
 			r.URL = u
+                        log.Printf("token before set scheme: %s", r.URL.Scheme)
+                        r.URL.Scheme = "https"
```
Comment (alextylor-e2x) [inline main.go]: check if setting the Scheme was required, as I think that was leftover from testing.

File: main.go [hunk 5]
```diff
@@ -216,6 +224,7 @@ func (rrt *registryRoundtripper) RoundTrip(req *http.Request) (*http.Response, e
 	}
 
 	resp, err := http.DefaultTransport.RoundTrip(req)
+          log.Printf("Response:\n%+v", resp)
```
Comment (alextylor-e2x) [inline main.go]: debugging so can remove

File: main.go [hunk 6]
```diff
@@ -145,19 +145,27 @@ func tokenProxyHandler(tokenEndpoint, repoPrefix string) http.HandlerFunc {
...
+                        log.Printf("this is the scope: %s", scope)
 			u, _ := url.Parse(tokenEndpoint)
 			u.RawQuery = q.Encode()
 			r.URL = u
+                        log.Printf("token before set scheme: %s", r.URL.Scheme)
+                        r.URL.Scheme = "https"
+                        log.Printf("token after set scheme: %s", r.URL.Scheme)
 			log.Printf("tokenProxyHandler: rewrote url:%s into:%s", orig, r.URL)
 			r.Host = u.Host
+                        log.Printf("\ntoken host: %s\n", r.Host)
```
Comment (alextylor-e2x) [inline main.go]: debugging so can remove

File: Dockerfile [hunk 7]
```diff
@@ -11,5 +11,4 @@ COPY --from=build /app /app
 # uncomment the following two lines if you're exposing a private GCR registry
 # COPY key.json /key.json
 # ENV GOOGLE_APPLICATION_CREDENTIALS /key.json
-
```
Comment (alextylor-e2x) [inline Dockerfile]: restore this new line, so we are changing the minimum amount

File: hadolint.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+ignored:
```
Comment (alextylor-e2x) [inline hadolint.yml:1]: can we add coments to outline what hte rules ignored are

File: .github/workflows/docker.yml [hunk 9]
```diff
@@ -0,0 +1,28 @@
+---
+name: Docker Verify
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
```
Comment (alextylor-e2x) [inline .github/workflows/docker.yml]: remove paths:... as infired already as all (i.e. your *)

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -1,26 +1,46 @@
 name: CI
 
 on:
-  pull_request:
   push:
     branches:
-      - master
+    - master
+    - proxy
```
Comment (alextylor-e2x) [inline .github/workflows/ci.yml]: note: to remove before merge, used for testing

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -1,26 +1,46 @@
...
+    - master
 
 jobs:
-  golangci:
-    name: lint
-    runs-on: ubuntu-latest
-    steps:
+    publish:
+      runs-on: ubuntu-latest
+      strategy:
```
Comment (alextylor-e2x) [inline .github/workflows/ci.yml]: remove matrix strategy, just hard code value

File: main.go [hunk 12]
```diff
@@ -145,19 +145,27 @@ func tokenProxyHandler(tokenEndpoint, repoPrefix string) http.HandlerFunc {
 		FlushInterval: -1,
```
Comment (alextylor-e2x) [inline main.go]: run `go fmt` against `main.go`!

File: main.go (hunk 1 above)
Comment (chaithra-shivaprasad) [reply main.go]: done

File: main.go (hunk 2 above)
Comment (chaithra-shivaprasad) [reply main.go]: done

File: main.go (hunk 3 above)
Comment (chaithra-shivaprasad) [reply main.go]: done

File: main.go (hunk 4 above)
Comment (chaithra-shivaprasad) [reply main.go]: done

File: main.go (hunk 5 above)
Comment (chaithra-shivaprasad) [reply main.go]: done

File: Dockerfile (hunk 7 above)
Comment (chaithra-shivaprasad) [reply Dockerfile]: done

File: hadolint.yml (hunk 8 above)
Comment (chaithra-shivaprasad) [reply hadolint.yml:1]: done

File: .github/workflows/docker.yml (hunk 9 above)
Comment (chaithra-shivaprasad) [reply .github/workflows/docker.yml]: done

File: .github/workflows/ci.yml (hunk 10 above)
Comment (chaithra-shivaprasad) [reply .github/workflows/ci.yml]: done

File: .github/workflows/ci.yml (hunk 11 above)
Comment (chaithra-shivaprasad) [reply .github/workflows/ci.yml]: done

**Review by alextylor-e2x — APPROVED** [review]

**Pull request merged**